### PR TITLE
Feature add metadata

### PR DESCRIPTION
**Pull request by christopheranderson**
Updated existing C# templates to have the various metadata files they need.

### CODE REVIEW

**Comment by christopheranderson** [comment]
Note: Node and the other languages haven't been completed, but I want to get this in so Naren can use these for his testing.

File: Templates/BlobTrigger-CSharp/sample.dat [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
+    "id":"foobar123"
+}
```
Comment (fabiocav) [inline Templates/BlobTrigger-CSharp/sample.dat]: Is this supposed to be here?

File: Templates/BlobTrigger/sample.dat [hunk 2]
```diff
@@ -0,0 +1,3 @@
+{
+    "id":"foobar123"
+}
```
Comment (fabiocav) [inline Templates/BlobTrigger/sample.dat]: Another one...

File: Templates/BlobTrigger-CSharp/sample.dat (hunk 1 above)
Comment (christopheranderson) [inline Templates/BlobTrigger-CSharp/sample.dat]: Yes. Maybe my sample data can be more clear it is a sample.

Sent from Outlook Mobile




On Mon, Mar 7, 2016 at 10:22 PM -0800, "Fabio Cavalcante" <notifications@github.com> wrote:





> @@ -0,0 +1,3 @@
> +{
> +    "id":"foobar123"
> +}

Is this supposed to be here?

---
Reply to this email directly or view it on GitHub:
https://github.com/Azure/azure-webjobs-sdk-templates/pull/7/files#r55320767

**Comment by alrod** [comment]
looks good to me

File: Templates/BlobTrigger-CSharp/readme.md [hunk 3]
```diff
@@ -0,0 +1,11 @@
+# BlobTrigger - C<span>#</span>
+
+The `BlobTrigger` makes it incredibly easy to react to new Blobs inside of Azure Blob Storage. This sample demostrates a simple use case of processing data from a given Blob using C#.
+
+## How it works
+
+For a `BlobTrigger` to work, you provide a path which dictates where the blobs are located inside your container, and can also help restrict the types of blobs you wish to return. For instance, you can set the path to `samples/{name}.png` to restrict the trigger to only the samples path and only blobs with ".png" at the end of their name.
```
Comment (mathewc) [inline Templates/BlobTrigger-CSharp/readme.md]: Might want to describe how the binding variables work (e.g. {name}), and how those captured values are available to other bindings. Not sure if/where we're describing how this works to users. Same for Queue trigger binding - we allow you to bind to any JSON properties in your message.

File: Templates/BlobTrigger/readme.md [hunk 4]
```diff
@@ -0,0 +1,11 @@
+# BlobTrigger - Node
+
+The `BlobTrigger` makes it incredibly easy to react to new Blobs inside of Azure Blob Storage. This sample demostrates a simple use case of processing data from a given Blob using Node.
```
Comment (mathewc) [inline Templates/BlobTrigger/readme.md]: "demostrates" spelled wrong. Perhaps we can run spell check on this stuff eventually? :)

File: Templates/BlobTrigger-CSharp/readme.md [hunk 5]
```diff
@@ -0,0 +1,11 @@
+# BlobTrigger - C<span>#</span>
+
+The `BlobTrigger` makes it incredibly easy to react to new Blobs inside of Azure Blob Storage. This sample demostrates a simple use case of processing data from a given Blob using C#.
```
Comment (mathewc) [inline Templates/BlobTrigger-CSharp/readme.md]: demostrates spelled wrong

File: Templates/HttpTrigger-CSharp/readme.md [hunk 6]
```diff
@@ -0,0 +1,11 @@
+# HttpTrigger - C<span>#</span>
+
+The `HttpTrigger` makes it incredibly easy to have your functions executed via an HTTP call to your function.
```
Comment (mathewc) [inline Templates/HttpTrigger-CSharp/readme.md]: fix wording. E.g. "...to have your functions executed via an HTTP call."

File: Templates/HttpTrigger-CSharp/readme.md [hunk 7]
```diff
@@ -0,0 +1,11 @@
...
+
+## How it works
+
+When you call the function, be sure you checkout which security rules you apply. If you're using an apikey, you'll need to include that in your request.
```
Comment (mathewc) [inline Templates/HttpTrigger-CSharp/readme.md]: "apikey" isn't our nomenclature is it? We have a "function key" that is specified in the url via a "code" query param, or x-functions-key header

File: Templates/TimerTrigger/readme.md [hunk 8]
```diff
@@ -0,0 +1,11 @@
+# TimerTrigger - Node
+
+The `TimerTrigger` makes it incredibly easy to have your functions executed on a schedule. This sample demostrates a simple use case of calling your function every 5 minutes.
```
Comment (mathewc) [inline Templates/TimerTrigger/readme.md]: looks like demostrates is mispelled everywhere. Recommend find/replace :)

**Pull request merged**